### ISSUE TITLE
Train and Test Split

### ISSUE THREAD

**Issue by rjtmehta99** [spro/practical-pytorch#124]
In the Char RNN classification I had some questions:
1. Why have not split the dataset into train and test/validation set to check the performance ?
2. How can we add 'accuracy' as a metric to evaluate model performance ?

**Comment by masalha-alaa** [comment]
I agree with this. Currently he tests on the training data, and draws a misleading confusion matrix. He trains on 100K names with repetitions, and tests against 10K from the same population, yet again with repetition. You can't infer accuracy from this. You need to split to train / test and evaluate on the test after every epoch. Please see [my notebook](https://github.com/masalha-alaa/name-classification-pytorch/blob/master/name_classification_rnn.ipynb).
